src/EFCore.PG/Design/Internal/NpgsqlAnnotationCodeGenerator.cs

```diff
@@ -13,8 +15,16 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Design.Internal
 {
     public class NpgsqlAnnotationCodeGenerator : AnnotationCodeGenerator
     {
-        public NpgsqlAnnotationCodeGenerator([NotNull] AnnotationCodeGeneratorDependencies dependencies)
-            : base(dependencies) {}
+        [CanBeNull] readonly Version _postgresVersion;
+
+        public NpgsqlAnnotationCodeGenerator(
+            [NotNull] AnnotationCodeGeneratorDependencies dependencies,
+            [NotNull] IDbContextOptions options)
```

Comment:
Is it possible to resolve `INpgsqlOptions` here?

Reply:
Good question - I looked into this. Inside EF Core you have two provider options types: SqlServerOptions and SqlServerOptionsExtension. The former is only used as a singleton option, i.e. as a way to make sure that contexts with different options get different singletons. Actual settings which affect the different services are looked up in SqlServerOptionsExtension. I'm assuming this is primarily to allow for options which don't cause separate singletons (maybe @ajcvickers can shed some more light).

BTW we have some other places where services consume NpgsqlOptions instead of NpgsqlOptionsExtension, we can clean that up later.

Reply:
Note - after some thought I ended up removing the version dependency for scaffolding. The idea is that we don't necessarily want the scaffolded database's version to actually impact or go into the scaffolded model, which may be used targeting a newer version etc.

Long story short, we now recognize both Serial and IdentityByDefault as by-convention strategies, which means they won't explicitly scaffolded into the model.